octoprint_smsnotifier/__init__.py

```diff
@@ -24,6 +151,11 @@ def get_settings_defaults(self):
             account_sid="",
             auth_token="",
             printer_name="",
+            #upload_provider="cloudinary",
+            upload_provider="aws_s3",
```

Comment:
You have this hardcoded correct? I will need to adjust the setting plugin page to account for this. I'll try to have it done by next weekend.

Reply:
Config.yml? I don't see it in the repo? I'm guessing it wont matter once I rebuild the config page. As you can see its been a couple weekends and I haven't taken a look at it yet.

Reply:
Octoprint's config.yaml:

```
plugins:
...
  smsnotifier:
    _config_version: 1
    account_sid: xxx
    auth_token: xxx
    body: '{printer_name} job complete after {elapsed_time}'
    enabled: true
    from_number: '1234'
    printer_name: e10
    recipient_number: '1234'
    upload_provider: aws_s3
    s3_bucket: dyn.tedder.me
    s3_key_prefix: 3d/automated/
    s3_url: https://dyn.tedder.me
    send_image: true
```

So twilio tokens are in the config, AWS credentials are in `~/.aws/credentials`.

It's been a while for me too :)

Reply:
oh you mean you can set plugin variables from the config.yml and not have to do it in the GUI?

Reply:
Exactly. Handy because I haven't modified the GUI to support all the necessary vars.

Reply:
ahh I was wondering how that would work. Well I think I can merge this into a test branch, and modify it back so cloudinary is the default and we can leave the other API backends as advanced use and make some documentation on how to set it up. We can leave the configuration page to another PR?

Reply:
Yeah, I like burying the switches. It makes me happy and yet it works by default.

I'll change the cloudinary thing and push it here, which'll be easier for you.

Reply:
=) thanks